eclass/kernel-install.eclass

```diff
@@ -459,15 +636,28 @@ kernel-install_install_all() {
 	local dir_ver=${1}
 	local kernel_dir=${EROOT}/usr/src/linux-${dir_ver}
 	local relfile=${kernel_dir}/include/config/kernel.release
+	local image_path=$(dist-kernel_get_image_path)
 	local module_ver
 	module_ver=$(<"${relfile}") || die
 
+	if [[ ${KERNEL_IUSE_GENERIC_UKI} ]]; then
+		if use generic-uki; then
+			# Populate placeholders
+			kernel-install_extract_from_uki linux "${kernel_dir}/${image_path}"
+			kernel-install_extract_from_uki initrd "${kernel_dir}/${image_path}"
```

Comment:
Oh, wait, that looks awfully confusing, sorry. Pass initrd path as 2nd arg instead, and remove the `out` subst from the `case` inside the function.

Reply:
Fixed, but I removed the case since now we are passing the name of the kernel image (which is variable) to the function instead of `linux`.